Assets/Scripts/Gameplay/ConnectionManagement/ConnectionManager.cs

```diff
@@ -0,0 +1,171 @@
+        }
+
+        void ApprovalCheck(byte[] connectionData, ulong clientId, NetworkManager.ConnectionApprovedDelegate connectionApprovedCallback)
+        {
+            m_CurrentState.ApprovalCheck(connectionData, clientId, connectionApprovedCallback);
+        }
+
+        public Task StartClientLobbyAsync(string playerName, Action<string> onFailure)
+        {
+            return m_CurrentState.StartClientLobbyAsync(playerName, onFailure);
+        }
+
+        public void StartClientIp(string playerName, string ipaddress, int port)
+        {
+            m_CurrentState.StartClientIP(playerName, ipaddress, port);
+        }
+
+        public void StartHostLobby(string playerName)
+        {
+            m_CurrentState.StartHostLobbyAsync(playerName);
+        }
+
+        public bool StartHostIp(string playerName, string ipaddress, int port)
+        {
+            return m_CurrentState.StartHostIP(playerName, ipaddress, port);
+        }
+
+        public void RequestShutdown()
+        {
+            m_CurrentState.OnUserRequestedShutdown();
+        }
+
+        /// <summary>
+        /// Registers the message handler for custom named messages. This should only be done once StartClient has been
+        /// called (start client will initialize NetworkSceneManager and CustomMessagingManager)
+        /// </summary>
+        public void RegisterCustomMessages()
+        {
+            NetworkManager.CustomMessagingManager.RegisterNamedMessageHandler(nameof(ReceiveServerToClientSetDisconnectReason_CustomMessage), ReceiveServerToClientSetDisconnectReason_CustomMessage);
+        }
+
+        void ReceiveServerToClientSetDisconnectReason_CustomMessage(ulong clientID, FastBufferReader reader)
+        {
+            reader.ReadValueSafe(out ConnectStatus status);
+            m_CurrentState.OnDisconnectReasonReceived(status);
+        }
+
+        /// <summary>
+        /// Sends a DisconnectReason to all connected clients. This should only be done on the server, prior to disconnecting the clients.
+        /// </summary>
+        /// <param name="status"> The reason for the upcoming disconnect.</param>
+        public static void SendServerToAllClientsSetDisconnectReason(ConnectStatus status)
+        {
+            var writer = new FastBufferWriter(sizeof(ConnectStatus), Allocator.Temp);
+            writer.WriteValueSafe(status);
+            NetworkManager.Singleton.CustomMessagingManager.SendNamedMessageToAll(nameof(ReceiveServerToClientSetDisconnectReason_CustomMessage), writer);
+        }
+
+        /// <summary>
+        /// Sends a DisconnectReason to the indicated client. This should only be done on the server, prior to disconnecting the client.
+        /// </summary>
+        /// <param name="clientID"> id of the client to send to </param>
+        /// <param name="status"> The reason for the upcoming disconnect.</param>
+        public static void SendServerToClientSetDisconnectReason(ulong clientID, ConnectStatus status)
```

Comment:
This is the same as above, just with a targeted client right? Why not merge them and have an optional parameter?

Reply:
with a "broadcast" parameter for example